be/src/exec/tablet_info.cpp

```diff
@@ -213,6 +214,27 @@ VOlapTablePartitionParam::VOlapTablePartitionParam(std::shared_ptr<OlapTableSche
         _partition_block.insert(
                 {slot->get_empty_mutable_column(), slot->get_data_type_ptr(), slot->col_name()});
     }
+
+    if (t_param.__isset.enable_automatic_partition && t_param.enable_automatic_partition) {
+        _is_auto_partiton = true;
+        DCHECK(t_param.__isset.partition_function_exprs);
+        //TODO: Is there a way to avoid the dependency of slot index?
+        DCHECK(t_param.partition_function_exprs[0].nodes[0].node_type ==
+               TExprNodeType::FUNCTION_CALL)
+                << t_param.partition_function_exprs[0].nodes[0].node_type;
+        DCHECK(t_param.partition_function_exprs[0].nodes[1].node_type == TExprNodeType::SLOT_REF)
+                << t_param.partition_function_exprs[0].nodes[1].node_type;
+        DCHECK(t_param.partition_function_exprs[0].nodes[2].node_type ==
+               TExprNodeType::STRING_LITERAL)
+                << t_param.partition_function_exprs[0].nodes[2].node_type;
+
+        vectorized::VExpr::create_expr_tree(t_param.partition_function_exprs[0],
+                                            _part_func_ctx); //RETURN IF ERROR
+        _partition_function = _part_func_ctx->root();
+        _part_interval =
```

Comment:
if not the data_trun() function. maybe do not have child(1） please recheck the logic

Reply:
will do in next pr. FE send the interval_slot_ref to BE.

Reply:
I think the work should do this PR， the way introduce lots of useless code

Reply:
Yes. fixed